### DISCUSSION TITLE
DatePicker - More Options

### DISCUSSION THREAD

**Discussion by w41x** [payloadcms/payload#4154] [Feature Requests & Ideas]
Since PayloadCMS ist using [react-datepicker](https://reactdatepicker.com) for the DateField, why only implement `dayAndTime`, `dayOnly`, `timeOnly`, and `monthOnly` for `date.pickerAppearance`, when there is also an option for `quarterOnly` (`showQuarterYearPicker`) and `yearOnly` (`showYearPicker`)?
Also, it would be nice to get access to some of the other options, e.g.: `showWeekNumbers`.
And in that regard maybe also a `weekOnly` option for `date.pickerAppearance`? But since this is not even a feature in `react-datepicker`, I doubt this would be as easy to implement as the other things.

Generally speaking one could implement a _**precision selection**_ (Year, Quarter, Month, Week, Day), but that's maybe overkill for the native DatePicker implementation and maybe something for a separate plugin.

I'm currently working with data, where the precision of dates often varies, i.e. for one document you know the exact date, for another document only the quarter. Thus I want to represent this with different 'precisions' to a given timestamp.

**Comment by DanRibbens** [comment]
Hey @w41x,

This could be a nice enhancement from the community if somebody wants to take it. I agree that sticking with the options that come with the component are first priority. If somebody wants a `weekOnly` that would need to be a custom component on top of the date type. For all the normal options, the input field itself may just work without any extra changes because we are spreading the date admin props to the DatePicker component. Still there will be a little work involved:

1. Update the DateField type to accomodate all the options from `react-datepicker`, you could probably map directly to the options DatePicker['property'] style of mapping.
1. Update the Joi field validation foir the config in `packages/payload/src/fields/config/schema.ts` to include any new options or changes to existing ones
1. The cell component would be nice to improve to take into account the different display formats based on the admin date properties:  packages/payload/src/admin/components/views/collections/List/Cell/field-types/Date/index.tsx
1. Add test properties to the date field test config `test/fields/collections/Date/index.ts`
1. Add some e2e tests for different date options
1. Update the `docs/fields/date.mdx` accordingly

It could be done fairly quickly I expect given the details here, nothing is too difficult here since it is all admin UI.

**Comment by aure1is** [comment]
Also, why is there no option for firstDayOfTheWeek?
